doc/src/guides/solving/solve-polynomial.md

```diff
@@ -0,0 +1,513 @@
+# Solve (Find the Roots of) a Polynomial Algebraically or Numerically 
+
+Use SymPy to solve (find the roots of) a univariate polynomial algebraically.
```

Comment:
Are there any multivariate polynomial solvers in SymPy or elsewhere?

Reply:
The standard way to solve multivariate polynomial systems is Gröbner bases, but using Gröbner bases is such a hairy and complicated thing, delving deep into algebraic geometry, that there is a _dedicated programming language_ for solving generic polynomial systems – [Singular](https://www.singular.uni-kl.de). For example, the exact polynomials in [my Shibuya repository](/Parcly-Taxel/Shibuya) were computed using Singular.

SymPy has a Gröbner basis and FGLM function, but it is too slow for most cases.

Reply:
> SymPy has a Gröbner basis and FGLM function, but it is too slow for most cases.

Well the plan is to make it faster by improving the algorithms and also by making it possible to use flint to speed up the polynomial arithmetic.

> Are there any multivariate polynomial solvers in SymPy or elsewhere?

In SymPy all of `solve`, `nonlinsolve` and `solve_poly_system` can solve simple systems of polynomial equations. The main limitation of all three is that they try to express solutions explicitly using radicals when actually the best symbolic representation of the solutions is usually the implicit one given by a Gröbner basis. The other limitation of all three is that they don't give a way to find or represent only the real solutions (which can easily be done from a Gröbner basis in the purely numeric case).

I think somewhere we need a guide that give a (very) simple explanation of Gröbner bases and how to use them to do the kinds of things that people want to do when they use solve. The problem here is that it's not really as simple as just saying "use this function to solve polynomial equations". Rather there needs to be some consideration of what it means to solve the equations and what the actual goal is. The question needs to be answered: what would you want to *do* with the solutions if I could give them to you?. The answer would then typically be that there is a better way to achieve the true end goal that bypasses having any explicit representation of the solutions.

Except in very simple cases demanding fully explicit representations of the solutions is misguided but the interfaces of `solve` et al don't provide any way to offer anything else. A multivariate version of `solveset` could do this easily but the danger would be that it would be just as hard to use as the sets currently returned by univariate `solveset`.

I'm still unsure what would be a good (i.e. useful) way to represent solutions of polynomial systems that users could understand and use without needing to know much about the theory of Gröbner bases.